src/ast.js

```diff
@@ -30,26 +30,31 @@ class Alternation {
  * Represents a character class (e.g. [0-9a-z])
  */
 class CharClass {
-  constructor(a, b) {
+  constructor(a, b, flags) {
     this.precedence = 1;
     this.set = regenerate(a, b);
+    this.flags = flags;
   }
 
   get length() {
     return 1;
   }
 
+  get isSingleCharacter() {
+    return !this.set.toArray().some(c => c > 0xffff);
```

Comment:
You may also need to modify `Literal#isSingleCharacter` as well to count code points instead of code units [here](https://github.com/devongovett/regexgen/blob/a5683bc6d4a2d84d76bfa792f967ec9852c31090/src/ast.js#L130). You can do that with `Array.from(this.value).length`.